sdk/src/types/block/context_input/reward.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::str::FromStr;
+
+use derive_more::From;
+
+use crate::types::block::Error;
+
+/// A Reward Input is an input that indicates which transaction Input is claiming Mana rewards.
+#[derive(Clone, Copy, Eq, PartialEq, Hash, Ord, PartialOrd, From, packable::Packable)]
+pub struct RewardContextInput(u16);
+
+impl RewardContextInput {
+    /// The context input kind of a [`RewardInput`].
+    pub const KIND: u8 = 2;
+
+    /// Creates a new [`RewardInput`].
+    pub fn new(index: u16) -> Self {
+        Self(index)
+    }
+
+    /// Returns the index of a [`RewardInput`].
+    pub fn index(&self) -> u16 {
+        self.0
+    }
+}
+
+#[cfg(feature = "serde")]
+string_serde_impl!(RewardContextInput);
```

Comment:
string impl for an u16?

Reply:
Also wasn't sure about that, don't we want this for all types?

Reply:
Only if they should serialise using their `Display` implementation